sections/tools/cypress-and-storybook-exposing-component-from-story.md

```diff
@@ -0,0 +1,221 @@
+    // initial state - popup is visible
+    ReactDOM.render(
+      <win.Popup showed={true} ref={popup} />,
+      win.document.querySelector(rootToMountSelector)
+    )
+  })
+
+  // act
+  cy.then(() => {
+    popup.current.hide()
+  })
+
+  // assert
+  cy.get(selectors.popupWindow).should('not.be.visible')
+})
+```
+
+## To sum it up: the roles of each participant
+
+Storybook:
+
+- Hosts Storybook Stories that contain bundled react components for test purposes.
+- Provides a real non-synthetic environment to run tests.
+- Each Story exposes one component in the global variable (to retrieve it in Cypress later).
+- Each Story exposes a component mount point (to mount a component in test).
+- Able to open each component in isolation in new tab.
+
+> Pro-Tip: Please, run another instance of Storybook for your component library or pages.
+
+Cypress:
+
+- Contains and runs tests and Javascript.
+- Visits isolated component Stories, retrieves component references from the global variable.
+- Renders component according to testing needs (with any data or test conditions such as mobile resolution).
+- Gives you super handy UI so you can see how your tests are going.
+
+## Conclusion
+
+Here I'd like to express my personal opinion and my colleagues' position about possible questions that may appear during the reading. Written below doesn't pretend to be true, may differ from reality and contain nuts.
+
+### My test utils use js-dom under the hood. Do I limit myself?
+
+- Js-dom is a synthetic environment. The separated DOM is not a real browser.
+- It doesn't really work out to act with js-dom as it user does. Especially when it comes to simulating input events.
+- How much confidence can you get from a written unit test if a component can be broken in CSS due to one incorrect z-index? If the component is tested by Cypress, you will see an error.
+- You write unit tests blindly. But why?
+
+### Should I choose the approach suggested?
+
+- If you use tests as a development environment - definitely, Yes!
+- If you look at tests as at **live** documentation - Yes.
+- If you really write unit-tests to cover things that too close to implementation and React-lifecycle - ... I don't know. I haven't been writing such a test for a long time. Are you sure that the covered logic is component responsibility? Maybe that logic should be extracted and tested accordingly?
+
+### Why not use cypress-react-unit-test then? Why do we need Storybook?
+
+Probably, it is our future to test components. There will be no need to maintain a separate instance of the Storybook, all tests will be entirely under the responsibility of Cypress, the configuration will be simplified, etc.
+But now the tool has [some](https://github.com/bahmutov/cypress-react-unit-test/issues/34) [problems](https://github.com/bahmutov/cypress-react-unit-test/issues/52) that make the environment provided incomplete for running tests. Hope that Gleb Bahmutov and the Cypress team will make it worked 🤞
```

Comment:
_"Cypress team will make it worked"_

Reply:
@muratkeremozcan what is the fix? Replacing the whole _Hope that Gleb Bahmutov and the Cypress team will make it worked 🤞_ sentence with _Cypress team will make it worked_?